rero_ils/modules/cli.py

```diff
@@ -1782,3 +1859,267 @@ def add_cover_urls(verbose):
+            output_directory, '{record_type}_metadata.csv'.format(
+                record_type=record_type
+            )
+        )
+        click.secho('\t{name}'.format(name=file_name_metadata), fg='green')
+        file_metadata = open(file_name_metadata, 'w')
+        file_name_ids = os.path.join(
+            output_directory, '{record_type}_ids.csv'.format(
+                record_type=record_type
+            )
+        )
+        click.secho('\t{name}'.format(name=file_name_ids), fg='green')
+        file_ids = open(file_name_ids, 'w')
+        file_name_errors = os.path.join(
+            output_directory, '{record_type}_errors.json'.format(
+                record_type=record_type
+            )
+        )
+        file_errors = open(file_name_errors, 'w')
+        file_errors.write('[')
+
+        for count, record in enumerate(records, 1):
+            pid = record.get('pid')
+            if create_pid:
+                pid = str(count)
+                record['pid'] = pid
+            uuid = str(uuid4())
+            if verbose:
+                click.secho(
+                    '{count}\t{record_type}\t{pid}:{uuid}'.format(
+                        count=count,
+                        record_type=record_type,
+                        pid=pid,
+                        uuid=uuid
+                    )
+                )
+            date = str(datetime.utcnow())
+            record['$schema'] = add_schema
+            if schema:
+                try:
+                    validate(record, schema)
+                    file_metadata.write(
+                        csv_metadata_line(record, uuid, date)
+                    )
+                    file_pidstore.write(
+                        csv_pidstore_line(record_type, pid, uuid, date)
+                    )
+                    file_ids.write(pid + '\n')
+                except Exception as err:
+                    click.secho(
+                        '{count}\t{record_type}: {msg}'.format(
+                            count=count,
+                            record_type=record_type,
+                            msg='Error validate in record: '
+                        ),
+                        fg='red')
+                    click.secho(str(err))
+                    if errors_count > 0:
+                        file_errors.write(',')
+                    errors_count += 1
+                    file_errors.write('\n')
+                    for line in json.dumps(record, indent=2).split('\n'):
+                        file_errors.write('  ' + line + '\n')
+
+        file_pidstore.close()
+        file_metadata.close()
+        file_ids.close()
+        file_errors.write('\n]')
+        file_errors.close()
+        if errors_count == 0:
+            os.remove(file_name_errors)
+    click.secho(
+        'Created: {count} Errors: {error_count}'.format(
+            count=count-errors_count,
+            error_count=errors_count
+        ),
+        fg='yellow'
+    )
```

Comment:
bad identation ? this `if` statement could/should be outside of the `with open(...) as infile` statement.

The `click.secho` statement could/should be the `else` statement of the same if (display the message only if errors are detected)

Reply:
ok